apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/modal.js

```diff
@@ -3,26 +3,72 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { Button, Modal } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { useEffect } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import React from 'react';
 import image from './image.svg';
 
 import './modal.scss';
 
 const GlobalStylesModal = () => {
+	const params = new URLSearchParams( window.location.search );
+	const [ showUnlockStylesModal, setShowUnlockStylesModal ] = useState(
+		params.get( 'unlock-styles' )
+	);
```

Comment:
Maybe we could use the onboarding state to change the copies of the modal? Something like using:
`	const onboard = select( ONBOARD_STORE ).getState();`
And check for the completed steps instead of relying on a query param? That way we wouldn't need changes in the  `controller.js` and `calypsoify-iframe.tsx` components.

Reply:
We cannot use the `ONBOARD_STORE` here unfortunately, since that only runs in Calypso.